core/chain_makers.go

```diff
@@ -105,6 +105,10 @@ func (b *BlockGen) AddTx(tx *types.Transaction) {
 	b.receipts = append(b.receipts, receipt)
 }
 
+func (b *BlockGen) AddReceipt(receipt *types.Receipt) {
+	b.receipts = append(b.receipts, receipt)
+}
```

Comment:
Can't we solve this somehow else? The chain maker currently generates valid, importable block chains. Injecting receipts directly makes the block invalid as the contents won't match the rest of the generated chain, so you won't be able to pass it through a block processor any more. Also, method docs are missing and it's very non trivial that this corrupts the chain.